documentation/FIR-MLIR-dialect.md

```diff
@@ -0,0 +1,204 @@
+# Investigating FIR as an MLIR Dialect
+
+## Introduction
+
+With the recent availability of the extensible MLIR framework as open-source, we've been investigating the potential to use MLIR as a substrate upon which to build a FIR dialect. This document will explain the motivations and the shape of this subproject using MLIR to define and implement a high-level operational representation of Fortran. (See also _Design: Fortran IR_ at [https://github.com/flang-compiler/f18/blob/master/documentation/FortranIR.md](https://github.com/flang-compiler/f18/blob/master/documentation/FortranIR.md) for more information.)
```

Comment:
80-character line length limit is being exceeded

Reply:
Yes. The markdown looks really bad with artificial line breaks added (at least in the viewer I used).

Reply:
Oh.  I use atom & it does a great job with line breaks in the source in its real-time preview window.  It even has a little line that shows you the 80-char limit on the screen.

Reply:
In case anyone had missed this, you can also see the markdown preview within the code review. On the top right corner of each file under the "files changed" tab, you can see three icons 
"Display the source diff"
"Display the rich diff"
"Options"

Display the rich diff shows the diff in markdown preview mode.

Reply:
Done